ServerCore/Pages/Puzzles/Feedback.cshtml

```diff
@@ -4,52 +4,79 @@
 @{
     ViewData["Title"] = "Feedback";
     var id = ViewContext.RouteData.Values["id"];
+    var role = ViewContext.RouteData.Values["eventRole"].ToString();
```

Comment:
You don't need to compute the role and do this if/else here. Instead you can put an attribute on the page, and in fact we've already put it on because we did a pass over all the pages. Merge and you will see it in the class file:

    [Authorize(Policy = "IsEventAdminOrAuthorOfPuzzle")]